diff --git a/duckdb_capi/duckdb.h b/duckdb_capi/duckdb.h
index a6cd9e2..a31eaea 100644
--- a/duckdb_capi/duckdb.h
+++ b/duckdb_capi/duckdb.h
@@ -787,7 +787,7 @@ process. Must be destroyed with 'duckdb_destroy_instance_cache'.
 
 * @return The database instance cache.
 */
-DUCKDB_C_API duckdb_instance_cache duckdb_create_instance_cache();
+DUCKDB_C_API duckdb_instance_cache duckdb_create_instance_cache(void);
 
 /*!
 Creates a new database instance in the instance cache, or retrieves an existing database instance.
@@ -924,7 +924,7 @@ Returns the version of the linked DuckDB, with a version postfix for dev version
 
 Usually used for developing C extensions that must return this for a compatibility check.
 */
-DUCKDB_C_API const char *duckdb_library_version();
+DUCKDB_C_API const char *duckdb_library_version(void);
 
 /*!
 Get the list of (fully qualified) table names of the query.
@@ -964,7 +964,7 @@ This should not be called in a loop as it internally loops over all the options.
 
 * @return The amount of config options available.
 */
-DUCKDB_C_API size_t duckdb_config_count();
+DUCKDB_C_API size_t duckdb_config_count(void);
 
 /*!
 Obtains a human-readable name and description of a specific configuration option. This can be used to e.g.
@@ -1499,7 +1499,7 @@ This is the amount of tuples that will fit into a data chunk created by `duckdb_
 
 * @return The vector size.
 */
-DUCKDB_C_API idx_t duckdb_vector_size();
+DUCKDB_C_API idx_t duckdb_vector_size(void);
 
 /*!
 Whether or not the duckdb_string_t value is inlined.
@@ -2786,7 +2786,7 @@ Creates a value of type SQLNULL.
 
 * @return The duckdb_value representing SQLNULL. This must be destroyed with `duckdb_destroy_value`.
 */
-DUCKDB_C_API duckdb_value duckdb_create_null_value();
+DUCKDB_C_API duckdb_value duckdb_create_null_value(void);
 
 /*!
 Returns the number of elements in a LIST value.
@@ -3445,7 +3445,7 @@ The return value must be destroyed with `duckdb_destroy_scalar_function`.
 
 * @return The scalar function object.
 */
-DUCKDB_C_API duckdb_scalar_function duckdb_create_scalar_function();
+DUCKDB_C_API duckdb_scalar_function duckdb_create_scalar_function(void);
 
 /*!
 Destroys the given scalar function object.
@@ -3711,7 +3711,7 @@ The return value should be destroyed with `duckdb_destroy_aggregate_function`.
 
 * @return The aggregate function object.
 */
-DUCKDB_C_API duckdb_aggregate_function duckdb_create_aggregate_function();
+DUCKDB_C_API duckdb_aggregate_function duckdb_create_aggregate_function(void);
 
 /*!
 Destroys the given aggregate function object.
@@ -3868,7 +3868,7 @@ The return value should be destroyed with `duckdb_destroy_table_function`.
 
 * @return The table function object.
 */
-DUCKDB_C_API duckdb_table_function duckdb_create_table_function();
+DUCKDB_C_API duckdb_table_function duckdb_create_table_function(void);
 
 /*!
 Destroys the given table function object.
@@ -5026,7 +5026,7 @@ Creates a new cast function object.
 
 * @return The cast function object.
 */
-DUCKDB_C_API duckdb_cast_function duckdb_create_cast_function();
+DUCKDB_C_API duckdb_cast_function duckdb_create_cast_function(void);
 
 /*!
 Sets the source type of the cast function.
diff --git a/duckdb_capi/duckdb_extension.h b/duckdb_capi/duckdb_extension.h
index 2d89b4b..287bb87 100644
--- a/duckdb_capi/duckdb_extension.h
+++ b/duckdb_capi/duckdb_extension.h
@@ -78,9 +78,9 @@ typedef struct {
 	void (*duckdb_interrupt)(duckdb_connection connection);
 	duckdb_query_progress_type (*duckdb_query_progress)(duckdb_connection connection);
 	void (*duckdb_disconnect)(duckdb_connection *connection);
-	const char *(*duckdb_library_version)();
+	const char *(*duckdb_library_version)(void);
 	duckdb_state (*duckdb_create_config)(duckdb_config *out_config);
-	size_t (*duckdb_config_count)();
+	size_t (*duckdb_config_count)(void);
 	duckdb_state (*duckdb_get_config_flag)(size_t index, const char **out_name, const char **out_description);
 	duckdb_state (*duckdb_set_config)(duckdb_config config, const char *name, const char *option);
 	void (*duckdb_destroy_config)(duckdb_config *config);
@@ -97,7 +97,7 @@ typedef struct {
 	duckdb_result_type (*duckdb_result_return_type)(duckdb_result result);
 	void *(*duckdb_malloc)(size_t size);
 	void (*duckdb_free)(void *ptr);
-	idx_t (*duckdb_vector_size)();
+	idx_t (*duckdb_vector_size)(void);
 	bool (*duckdb_string_is_inlined)(duckdb_string_t string);
 	uint32_t (*duckdb_string_t_length)(duckdb_string_t string);
 	const char *(*duckdb_string_t_data)(duckdb_string_t *string);
@@ -235,7 +235,7 @@ typedef struct {
 	duckdb_value (*duckdb_get_map_key)(duckdb_value value, idx_t index);
 	duckdb_value (*duckdb_get_map_value)(duckdb_value value, idx_t index);
 	bool (*duckdb_is_null_value)(duckdb_value value);
-	duckdb_value (*duckdb_create_null_value)();
+	duckdb_value (*duckdb_create_null_value)(void);
 	idx_t (*duckdb_get_list_size)(duckdb_value value);
 	duckdb_value (*duckdb_get_list_child)(duckdb_value value, idx_t index);
 	duckdb_value (*duckdb_create_enum_value)(duckdb_logical_type type, uint64_t value);
@@ -297,7 +297,7 @@ typedef struct {
 	void (*duckdb_validity_set_row_validity)(uint64_t *validity, idx_t row, bool valid);
 	void (*duckdb_validity_set_row_invalid)(uint64_t *validity, idx_t row);
 	void (*duckdb_validity_set_row_valid)(uint64_t *validity, idx_t row);
-	duckdb_scalar_function (*duckdb_create_scalar_function)();
+	duckdb_scalar_function (*duckdb_create_scalar_function)(void);
 	void (*duckdb_destroy_scalar_function)(duckdb_scalar_function *scalar_function);
 	void (*duckdb_scalar_function_set_name)(duckdb_scalar_function scalar_function, const char *name);
 	void (*duckdb_scalar_function_set_varargs)(duckdb_scalar_function scalar_function, duckdb_logical_type type);
@@ -316,7 +316,7 @@ typedef struct {
 	void (*duckdb_destroy_scalar_function_set)(duckdb_scalar_function_set *scalar_function_set);
 	duckdb_state (*duckdb_add_scalar_function_to_set)(duckdb_scalar_function_set set, duckdb_scalar_function function);
 	duckdb_state (*duckdb_register_scalar_function_set)(duckdb_connection con, duckdb_scalar_function_set set);
-	duckdb_aggregate_function (*duckdb_create_aggregate_function)();
+	duckdb_aggregate_function (*duckdb_create_aggregate_function)(void);
 	void (*duckdb_destroy_aggregate_function)(duckdb_aggregate_function *aggregate_function);
 	void (*duckdb_aggregate_function_set_name)(duckdb_aggregate_function aggregate_function, const char *name);
 	void (*duckdb_aggregate_function_add_parameter)(duckdb_aggregate_function aggregate_function,
@@ -343,7 +343,7 @@ typedef struct {
 	duckdb_state (*duckdb_add_aggregate_function_to_set)(duckdb_aggregate_function_set set,
 	                                                     duckdb_aggregate_function function);
 	duckdb_state (*duckdb_register_aggregate_function_set)(duckdb_connection con, duckdb_aggregate_function_set set);
-	duckdb_table_function (*duckdb_create_table_function)();
+	duckdb_table_function (*duckdb_create_table_function)(void);
 	void (*duckdb_destroy_table_function)(duckdb_table_function *table_function);
 	void (*duckdb_table_function_set_name)(duckdb_table_function table_function, const char *name);
 	void (*duckdb_table_function_add_parameter)(duckdb_table_function table_function, duckdb_logical_type type);
@@ -417,7 +417,7 @@ typedef struct {
 	void (*duckdb_destroy_task_state)(duckdb_task_state state);
 	bool (*duckdb_execution_is_finished)(duckdb_connection con);
 	duckdb_data_chunk (*duckdb_fetch_chunk)(duckdb_result result);
-	duckdb_cast_function (*duckdb_create_cast_function)();
+	duckdb_cast_function (*duckdb_create_cast_function)(void);
 	void (*duckdb_cast_function_set_source_type)(duckdb_cast_function cast_function, duckdb_logical_type source_type);
 	void (*duckdb_cast_function_set_target_type)(duckdb_cast_function cast_function, duckdb_logical_type target_type);
 	void (*duckdb_cast_function_set_implicit_cast_cost)(duckdb_cast_function cast_function, int64_t cost);
@@ -529,7 +529,7 @@ typedef struct {
 
 // Exposing the instance cache
 #ifdef DUCKDB_EXTENSION_API_VERSION_UNSTABLE
-	duckdb_instance_cache (*duckdb_create_instance_cache)();
+	duckdb_instance_cache (*duckdb_create_instance_cache)(void);
 	duckdb_state (*duckdb_get_or_create_from_cache)(duckdb_instance_cache instance_cache, const char *path,
 	                                                duckdb_database *out_database, duckdb_config config,
 	                                                char **out_error);
